### PR TITLE
[ICDS] Exclude test data in aggregation

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @calellowitz, @Rohit25negi, @snopoke  

I added ```[location_type]_is_test``` property to the rest of the aggregation scripts where we need this. Also, I excluded test records when we calculate data for parent locations.

This PR is open to review but please don't merge it, I'm waiting for merge this one https://github.com/dimagi/commcare-hq/pull/22492 and update a migration file  

Feel free to add comments.

Regards,
Łukasz

### CODE REVIEW

**Comment by Rohit25negi** [comment]
test case failure looks related and needs to be addressed
```
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/icds_reports/tasks.py", line 293, in icds_aggregation_task
    func(date)
  File "/mnt/commcare-hq-ro/custom/icds_reports/utils/__init__.py", line 740, in wrapper
    result = func(*args, **kwargs)
  File "/mnt/commcare-hq-ro/custom/icds_reports/tasks.py", line 478, in _agg_child_health_table
    AggChildHealth.aggregate(force_to_date(day))
  File "/mnt/commcare-hq-ro/custom/icds_reports/models/aggregate.py", line 635, in aggregate
    cursor.execute(query)
  File "/vendor/lib/python2.7/site-packages/raven/contrib/django/client.py", line 114, in execute
    return real_execute(self, sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
ProgrammingError: operator does not exist: smallint == integer
LINE 7:             WHERE awc_is_test == 0

```

**Comment by lwyszomi** [comment]
@Rohit25negi @calellowitz @snopoke I updated PR

**Comment by lwyszomi** [comment]
Hi @Rohit25negi @calellowitz @snopoke,

I updated migration file so now PR is also ready for merge. Please let me know if you have any questions.

@Rohit25negi once again thank you that you merged PR which blocking this one 😄 

Regards,
Łukasz

**Comment by calellowitz** [comment]
@lwyszomi looks there is a merge conflict here

**Comment by lwyszomi** [comment]
@calellowitz updated

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: nit: this will reassign the `child_location` multiple times if `aggregation_level` is greater than 2, once for each level. if you reversed the order of the conditions it will only reassign once

File: custom/icds_reports/utils/aggregation_helpers/agg_awc_daily.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc_daily.py]: same as above

File: custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py]: why do you change the group_by here

File: custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py]: same comment on ordering

File: custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_ccs_record.py]: Nevermind, i see you fixed it

**Review by calellowitz — APPROVED** [review]

**Comment by calellowitz** [comment]
@lwyszomi One comment and it is not blocking, but it does apply to each aggregation

**Review by calellowitz — CHANGES_REQUESTED** [review]
Can you add tests for this as well? I think it can be as simple as changing some of the new fixture values to be `1` and making sure those propagate as expected. Should probably test at all levels

**Review by calellowitz — APPROVED** [review]

File: custom/icds_reports/tests/reports/test_children_initiated.py
Comment (nickpell) [inline custom/icds_reports/tests/reports/test_children_initiated.py]: @lwyszomi Can you remove these print statements from tests?

**Pull request merged**
